### PR TITLE
Updated NPCs

### PR DESCRIPTION
**Pull request by Hirasu**
Updated NPCs with comments.
I removed all tree root crystals, triggers and spawn locations for now. If someone does a spawn system I could make all crystal/monster spawn locations for every map. It worked very well in the past with @gastonb spawn system.
https://github.com/gastonb/melia/commit/9e8e0069e1a7f2b95d981a4d0cf17c6c5bbe20db

I'll re-add the triggers on an another pull request.

### CODE REVIEW

**Comment by Hirasu** [comment]
@exectails Should I add the dialogs?
https://github.com/celophi/melia/blob/master/system/scripts/npc/city/c_Klaipe.lua

**Comment by exectails** [comment]
What made you decide to include information about the quests, and where everything came from? I imagine that might be helpful to someone who's digging around the files as well, but to others it might be just a lot of noise in the scripts.
```lua
+--- npcselectdialog.ies | ClassName: MASTER_DOPPELSOELDNER ---
 +--- QuestIDs: JOB_DOPPELSOELDNER5_1 | JOB_DOPPELSOELDNER_6_1 | JOB_DOPPELSOELDNER_8_1 ---
```

>Should I add the dialogs?

I would say yes, but are you certain that they're correct? This for example
````
function npc_ETC_20150717_013603()
msg("KLAPEDA_Emilia_basic22")
msg("JOB_SAPPER2_2_prog1")
msg("JOB_DIEVDIRBYS3_1_succ1")
msg("JOB_DIEVDIRBYS3_2_select1")
msg("JOB_DIEVDIRBYS3_2_agree1")
msg("JOB_DIEVDIRBYS3_2_succ1")
msg("JOB_DIEVDIRBYS3_3_select1")
msg("Emilia_Select_1")
msg("EAST_PREPARE_COMP")
msg("Emilia_Select_3")
msg("Emilia_Select_2")
msg("EAST_PREPARE_1_ST")
msg("EAST_PREPARE_1_AC")
msg("EAST_PREPARE_1_SUCST")
end
```
looks a little weird to me. Especially those "select"s and "agree"s in there. Is that really a linear dialog of X messages, or should there be selects? Btw, you should format the scripts properly, to make reading them easier and to conform to the coding conventions.

The other thing to consider is #103, with the problem being highlighted very well in the above example. You have no idea what's going on there, you can barely even tell what NPC that even is. What's your opinion on that issue, as the only one who's currently working on scripts?

And while we're on the topic of issues, #112 might affect you and your scripts as well.

**Comment by Hirasu** [comment]
Yes they should have selections(not all). I got all information from the client and I could "reproduce" the quest line with this informations.

Example: 
--- [Cleric Master] Rozalija --- 
--- npcselectdialog.ies | ClassName: MASTER_CLERIC ---

npcselectdialog.ies "Classname" MASTER_CLERIC
Now you can see all questID from QuestName1 to QuestName50.

But you have to check if the NPC is the Start/Prog/End NPC of the quest, look to the questprogresscheck.ies and search for the JOB_CLERIC1 (QuestName1):

 StartDesc: 클레릭 마스터와 대화
QUEST_JOBSTEP_20150317_001416
"Talk to the Cleric Master"

StartStory: 클라페다의 클레릭 마스터와 대화하세요.
QUEST_JOBSTEP_20150317_001658
"Talk to the Cleric Master in Klaipeda."

StartLocation: c_Klaipe MASTER_CLERIC 125
StartMap: c_Klaipe
StartNPC: MASTER_CLERIC

I can remove the questID because the ClassName is enough to get all information about this NPC.
About **https://github.com/aura-project/melia/issues/103** I like the idea and maybe I should wait with the dialogs until its decided.

**Comment by exectails** [comment]
I guess we have different approaches to NPCs and quests, I would always go on officials and log them, that way I can be sure the data is 100% correct and up-to-date, while also not having to worry about decrypting their weird table scripting system^^

>Yes they should have selections(not all)

Why didn't you include the selects if they *should* be there? I could imagine the dialog might be a little weird without them.

**Comment by exectails** [comment]
Any news on this?

**Comment by Hirasu** [comment]
Sorry I was busy the last days.
> I guess we have different approaches to NPCs and quests, I would always go on officials and log them, that way I can be sure the data is 100% correct and up-to-date, while potentially getting additional information/packets and also not having to worry about decrypting their weird table scripting system

This should be the best way, maybe 'someone' can write a plugin with PaleTree for that. I saw something like that in the gitter chat (1-2 years ago).

I'll remove all comments except the npc names so that we have updated npc locations for now.

**Comment by exectails** [comment]
You have to resolve the outstanding conflict for me to merge it.

File: system/scripts/npc/city/c_Klaipe.lua [hunk 1]
```diff
@@ -1,216 +1,213 @@
---- [Item Merchant] Mirina --- 
+--- [Item Merchant] Mirina ---
 addnpc(20115, "ETC_20150717_013603", "c_Klaipe", 511, -1, -349, 45, "npc_itemmerchantmirina")
 function npc_itemmerchantmirina()
-	msg("Emilia_Select_1")
-	openshop("Klapeda_Misc")
+    msg("Emilia_Select_1")
```
Comment (exectails) [inline system/scripts/npc/city/c_Klaipe.lua]: You should be using tabs, as per our coding conventions.

File: system/scripts/npc/city/c_Klaipe.lua [hunk 2]
```diff
@@ -1,216 +1,213 @@
...
+--- Refugee ---
 addnpc(20139, "QUEST_LV_0100_20150317_001418", "c_Klaipe", -60, 79, -446, -33.69, "npc_dummy")
-
---- [Market Manager] Logi --- 
+ 
+--- [Market Manager] Logi ---
 addnpc(20169, "ETC_20150317_009211", "c_Klaipe", 278, 79, 137, 45, "npc_dummy")
-
---- [Peltasta Master]Maria Leed --- 
+
```
Comment (exectails) [inline system/scripts/npc/city/c_Klaipe.lua]: Without those useless spaces this diff would be much shorter^^

Comment (Hirasu) [reply system/scripts/npc/city/c_Klaipe.lua]: I'll remove the spaces.

File: system/scripts/npc/city/c_Klaipe.lua (hunk 1 above)
Comment (Hirasu) [reply system/scripts/npc/city/c_Klaipe.lua]: Sorry...

**Comment by exectails** [comment]
Still waiting for the addition of the Swordsman Master NPC.

**Comment by exectails** [comment]
Thanks for the update.

**Pull request merged**